go/hyperkit.go

```diff
@@ -544,3 +548,14 @@ func getHome() string {
 	}
 	return os.Getenv("HOME")
 }
+
+// from http://stackoverflow.com/questions/24153268/how-do-i-write-an-async-child-reaper-in-go
+func reapChild(cmd *exec.Cmd) {
+	sigc := make(chan os.Signal, 1)
+	signal.Notify(sigc, syscall.SIGCHLD)
+	go func() {
+		_ = <-sigc
+		cmd.Wait()
+		signal.Stop(sigc)
+	}()
+}
```

Comment:
Can't you just do `go cmd.Wait()` at the top level (where you call `reapChild` here) to achieve the same thing?

If the process has multiple children and any one of them other than `cmd` exits then this code basically degrades into that anyway.

Otherwise you need a for loop and `wait4` with `WNOHANG` to keep handling `SIGCHLD` and doing a non-blocking check if it was the one you were interested in until it is.